dot/state/block.go

```diff
@@ -104,8 +103,8 @@ func NewBlockState(db chaindb.Database, trs *Tries, telemetry telemetry.Client)
 
 // NewBlockStateFromGenesis initialises a BlockState from a genesis header,
 // saving it to the database located at basePath
-func NewBlockStateFromGenesis(db chaindb.Database, trs *Tries, header *types.Header,
-	telemetryMailer telemetry.Client) (*BlockState, error) {
+func NewBlockStateFromGenesis(db *chaindb.BadgerDB, trs *Tries, header *types.Header,
```

Comment:
This is left as an implementation on purpose, since it should be changed by issue #2981